python/ray/tests/accelerators/test_npu.py

```diff
@@ -0,0 +1,125 @@
+import os
+import sys
+import pytest
+from unittest.mock import patch
+
+import ray
+from ray._private.accelerators import NPUAcceleratorManager as Accelerator
+
+
+@patch("glob.glob")
+@patch("os.listdir")
+def test_autodetect_num_npus(mock_list, mock_glob):
+    mock_glob.return_value = [f"/dev/davinci{i}" for i in range(4)]
+    mock_list.return_value = []
+    assert Accelerator.get_current_node_num_accelerators() == 4
+
+
+@patch("glob.glob")
+@patch("os.listdir")
+def test_autodetect_num_npus_without_devices(mock_list, mock_glob):
+    mock_glob.side_effect = FileNotFoundError
+    mock_list.return_value = []
+    assert Accelerator.get_current_node_num_accelerators() == 0
+
+
+def test_ascend_npu_accelerator_manager_api():
+    assert Accelerator.get_resource_name() == "NPU"
+    assert Accelerator.get_visible_accelerator_ids_env_var() == "ASCEND_VISIBLE_DEVICES"
+    assert Accelerator.validate_resource_request_quantity(0.5) == (True, None)
+    assert Accelerator.validate_resource_request_quantity(1) == (True, None)
+
+
+def test_visible_ascend_npu_type(shutdown_only):
+    with patch.object(
+        Accelerator, "get_current_node_num_accelerators", return_value=4
+    ), patch.object(
+        Accelerator, "get_current_node_accelerator_type", return_value="Ascend910B"
+    ):
+        os.environ["ASCEND_VISIBLE_DEVICES"] = "0,1,2"
```

Comment:
You can use monkeypatch to change the `os.environ` so that it will be auto reverted when the test finishes.

Reply:
Great suggestions, I've made the changes.